activerecord/lib/active_record/relation/query_methods.rb

```diff
@@ -948,6 +948,25 @@ def skip_preloading! # :nodoc:
       self
     end
 
+    # Adds an SQL comment to queries generated from this relation. For example:
+    #
+    #   User.annotate("selecting user names").select(:name)
+    #   # SELECT "users"."name" FROM "users" /* selecting user names */
+    #
+    # The SQL block comment delimiters, "/*" and "*/", will be added automatically.
+    def annotate(comment)
```

Comment:
This looks odd to me. All multi value query methods defined by `def multi_value_method(*args)` except the `annotate`.

Reply:
This is a little outside the norm for multi-value methods, but I believe this is a necessary compromise between the [desired](https://github.com/rails/rails/pull/35617#discussion_r266260622) API for `annotate` and the requirements of working with [`Relation::Merger`](https://github.com/rails/rails/blob/9c1adcaaf7ec27b5f994fe119c8a56516a514bea/activerecord/lib/active_record/relation/merger.rb#L76), e.g., to pass [this test](https://github.com/rails/rails/blob/9c1adcaaf7ec27b5f994fe119c8a56516a514bea/activerecord/test/cases/relation/merging_test.rb#L139).

Reply:
What is the desired behavior for `User.annotate("foo").annotate("bar")`?

Reply:
The desired result would be a single SQL comment, `/* foo bar */`.

This is inline with how `#optimizer_hints` includes each hint in a single block, though of course that's required by the way hints are implemented across databases.

The alternative would be `/* foo */ /* bar */`. Do you think that is preferable?

Reply:
Would be a single SQL comment is fine for now (merging different relation comments `Post.annotate("post comment").joins(:tags).merge(Tag.annotate("tag comment")) #=> "/* post comment tag comment */"` looks odd to me though).

My point is that, if defined by `def annotate(comment)`, it reminds me of single value method (i.e. `User.annotate("foo").annotate("bar") # => /* bar */`, that is the reason for the my confusion https://github.com/rails/rails/pull/35617#discussion_r267547547).

Since the `def annotate(comment)` makes any other confusion, it would be better to allow multiple values as multiple values query method like other multiple values query methods (`select`, `group`, etc.) for consistency for now.

Reply:
I'm fine with `User.annotate("foo").annotate("bar") # => /* foo bar */`.

But if it behaves so, the single value method like method definition makes me (and any other people) confused that the `annotate` is single value method (i.e. `User.annotate("foo").annotate("bar") # => /* bar */`, that is the reason for the my confusion https://github.com/rails/rails/pull/35617#discussion_r267547547).

As a multiple values method, the `select` behaves `User.select("foo").select("bar") # => SELECT "users"."foo", "users"."bar" ...`, so as the same multiple values method, it is worth to be defined the `annotate` in the same way.

Can you change it back to be defined in the multiple values method way?

cc @jeremy